### DISCUSSION TITLE
Docker container with IPv6 addresses

### DISCUSSION THREAD

**Discussion by MaZe3D** [shlinkio/discussions#88] [Help wanted] [answered]
#### How Shlink is set-up

* Shlink Version: (latest Docker image)
* PHP Version: (latest Docker image)
* How do you serve Shlink: Docker image
* Database engine used: MariaDB

#### Summary

Hello everyone, I tried to setup shlink and tried it by providing the docker-container a IPv6 address. The problem is, that the Docker-Image can only be accessed via IPv4. 
Now my question is, weather I did't set up the docker correctly or the service within the docker currently cannot communicate via IPv6. If the last one is the case, maybe you can help me figureing out which setting to change inside the docker to get around this problem. 
The IPv6 configuration is working as I am able to Ping the container but cant access it (over IPv6) via the browser.

Thank you for your time and have a loveley day!

**Comment by ikuradon** [comment, marked as answer]
My workaround on IPv6 only kubernetes cluster, override config/autoload/swoole.global.php with:
```
<?php

declare(strict_types=1);

use Shlinkio\Shlink\Core\Config\EnvVars;

use const Shlinkio\Shlink\MIN_TASK_WORKERS;

return (static function (): array {
    $taskWorkers = (int) EnvVars::TASK_WORKER_NUM->loadFromEnv(16);

    return [

        'mezzio-swoole' => [
            // Setting this to true can have unexpected behaviors when running several concurrent slow DB queries
            'enable_coroutine' => false,

            'swoole-http-server' => [
                'host' => '[::]',
                'port' => (int) EnvVars::PORT->loadFromEnv(8080),
                'process-name' => 'shlink',
                'protocol' => SWOOLE_SOCK_TCP6,

                'options' => [
                    'worker_num' => (int) EnvVars::WEB_WORKER_NUM->loadFromEnv(16),
                    'task_worker_num' => max($taskWorkers, MIN_TASK_WORKERS),
                ],
            ],
        ],

    ];
})();
```
The changes include:
* host `0.0.0.0` to `[::]`
* Add protocol `SWOOLE_SOCK_TCP6` to support IPv6.

_Not checked how IPv4-only docker will affect._
